feat: (RandomCoin ) Add RandomCoin component to AdBanner

amirxhero opened the pull request:
Introduces a new RandomCoin component that displays a clickable coin animation when ads are loaded in AdBanner. Also adds the coin image asset and updates AdBanner.vue to conditionally render RandomCoin based on ad load status,

amirxhero committed:
feat (RandomCoin ): Add RandomCoin component to AdBanner

Introduces a new RandomCoin component that displays a clickable coin animation when ads are loaded in AdBanner. Also adds the coin image asset and updates AdBanner.vue to conditionally render RandomCoin based on ad load status,

amirxhero committed:
fix (components) : Add Lottie animation to RandomCoin component

Replaced static coin image in RandomCoin.vue with a Lottie animation using @lottiefiles/dotlottie-vue and a new data.json asset. Updated AdBanner.vue to use lazy loading for the coin component. Added @lottiefiles/dotlottie-vue dependency and removed unused Google AdSense and VueUse packages.

amirxhero commented:
> Use this Lottie animation in the project instead of the previous coin image. You have full control to rotate it. Initially, show only the coin’s from side to make it harder for the user to notice. When the user approaches, rotate it 320 degrees so they can see it clearly. [coin motion.zip](https://github.com/user-attachments/files/21769892/coin.motion.zip)

Done.

amirxhero committed:
refactor(components): Refactor RandomCoin to support multiple coins

Updated RandomCoin.vue to display multiple coins with randomized types and positions, using TransitionGroup for animation. Added bronze, silver, and gold coin assets, and refactored logic for coin click handling and visibility checks. Also updated AdBanner.vue to always show the RandomCoin component.

sanaderi commented on public/images/random-coin.png:
Plz remove this one @amirxhero

sanaderi commented on components/common/RandomCoin.vue:
```diff
@@ -0,0 +1,185 @@
+
+<script setup>
+import { DotLottieVue } from '@lottiefiles/dotlottie-vue'
+import successSound from '@/assets/sounds/success.mp3'
```
<img width="1445" height="409" alt="image" src="https://github.com/user-attachments/assets/35f600f6-808f-4237-94fe-df22bd71f816" />

amirxhero committed:
Merge remote-tracking branch 'upstream/upgrade-v3' into 503-random-coin-route

amirxhero committed:
Refactor AdBanner.vue and remove unused image

Standardized string quotes and improved code readability in AdBanner.vue by switching to single quotes and removing unnecessary semicolons. Also deleted the unused random-coin.png image from the public/images directory.

amirxhero committed:
Update RandomCoin.vue

amirxhero committed:
refactor(components): Add v-once directive to lazy-common-random-coin

The v-once directive is added to the lazy-common-random-coin component to ensure it is rendered only once, improving performance and preventing unnecessary re-renders.

amirxhero committed:
refactor (components) : Refactor RandomCoin logic and relocate component

Moved the RandomCoin component from AdBanner.vue to layouts/default.vue for better placement. Refactored RandomCoin.vue to fetch coin types from an API, map coin types to sources, and post points on coin click. Improved coin animation logic and added error handling for API calls.
